CONTRIBUTING.md

```diff
@@ -0,0 +1,250 @@
+# Contributing
+
+- [Contributing](#contributing)
+  - [Prerequisites](#prerequisites)
+  - [Getting started](#getting-started)
+  - [Creating a new Component](#creating-a-new-component)
+    - [Rigid VS. Flexible](#rigid-vs-flexible)
+    - [Introducing 3rd Party Components](#introducing-3rd-party-components)
+    - [Moving A Component From `dcos/dcos-ui` To The `ui-kit`](#moving-a-component-from-dcos-dcos-ui-to-the-ui-kit)
+    - [Creating a Reusable Component](#creating-a-reusable-component)
+      - [Avoid Weak Elements](#avoid-weak-elements)
+      - [Declare Prop Types](#declare-prop-types)
+      - [Never hard code HTML ID’s](#never-hard-code-html-ids)
+      - [Use Logical Defaults](#use-logical-defaults)
+      - [A11y Support](#a11y-support)
+      - [Configuration Objects](#configuration-objects)
+      - [Single Responsibility Principle](#single-responsibility-principle)
+      - [Keep Components Simple](#keep-components-simple)
+      - [Directory Structure](#directory-structure)
+  - [Atomic Design](#atomic-design)
+    - [Atoms](#atoms)
+    - [Molecules](#molecules)
+    - [Organisms](#organisms)
+  - [Commit types](#commit-types)
+
+## Prerequisites
+
+We expect the user to be able to handle:
+
+* git
+* node
+* npm
+
+We expect a entry level of react knowledge, if you know these terms in a react context you are well prepared:
+
+* Component
+
+Also [here](https://reactjs.org/tutorial/tutorial.html) you can find a thorough introduction to react.
+
+We are using TypeScript which is a superset of JavaScript, you can get introduction [here](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html)
+
+## Getting started
+
+1.  Clone the repository and enter the directory.
+2.  run `npm install` and `npm run setup:git-hooks`
```

Comment:
with node 9.11.0 I get the following error. Works fine with node 8
```
> node@9.11.0 preinstall 
> node installArchSpecificPackage

module.js:471
    throw err;
    ^

Error: Cannot find module 'node-bin-setup'
    at Function.Module._resolveFilename (module.js:469:15)
```

Reply:
Works for me: 
![install](https://user-images.githubusercontent.com/156010/39580305-26b0a782-4ee9-11e8-9942-396c4f40eaa8.gif)

Reply:
😢 except jest, jest never finishes